torch/csrc/autograd/engine.cpp

```diff
@@ -237,6 +237,16 @@ size_t ReadyQueue::size() const {
   return heap_.size();
 }
 
+auto ReadyQueue::waitForWork() -> void {
+  {
+    // Lock mutex for accesses to heap_
+    std::unique_lock<std::mutex> lock(mutex_);
+    not_empty_.wait(lock, [this] { return !heap_.empty(); });
```

Comment:
would this cause the threads to spin, waiting for the work that'll never come?

Reply:
The wait on the condition variable is not a busy wait. It will just wait on the signal.